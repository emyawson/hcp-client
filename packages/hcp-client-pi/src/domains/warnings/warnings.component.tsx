import * as React from 'react';

import { SegmentedButton, TrendGraph } from '@roche/patterns-indicators/components';

import { trendGraphMockProps } from './warnings.mock';
import { WarningsHeader, WarningsWrapper } from './warnings.style';

type WarningsProps = {
  selectedSegment: string;
  selectSegment: (string) => void;
};

export const Warnings: React.StatelessComponent<WarningsProps> = ({
  selectedSegment,
  selectSegment,
}: WarningsProps) => (
  <WarningsWrapper>
    <WarningsHeader>
      <SegmentedButton
        selectedSegment={selectedSegment}
        labels={['Trend Graph', 'Log Book']}
        // tslint:disable-next-line:jsx-no-lambda
        onSegmentClick={clickedLabel => {
          selectSegment(clickedLabel);
        }}
      />
    </WarningsHeader>
    {selectedSegment === 'Trend Graph' && (
      <TrendGraph {...trendGraphMockProps} />
    )}
    {selectedSegment === 'Log Book' && 'Log Book'}
  </WarningsWrapper>
);
