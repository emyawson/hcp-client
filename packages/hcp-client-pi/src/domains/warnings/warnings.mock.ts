export const trendGraphMockProps = {
  verticalLabel: [{ value: 0.5, label: 'Blood Glucose (mg/dL)' }],
  verticalTicks: [
    {
      value: 0,
      label: '0',
    },
    {
      value: 0.25,
      label: '100',
    },
    {
      value: 0.5,
      label: '200',
    },
    {
      value: 0.75,
      label: '300',
    },
    {
      value: 1,
      label: '400',
    },
  ],
  horizontalDayTicks: [
    { value: 0, label: '18', isWeekend: false, drawLongTick: false },
    {
      value: 0.07142857142857142,
      label: '19',
      isWeekend: false,
      drawLongTick: false,
    },
    {
      value: 0.14285714285714285,
      label: '20',
      isWeekend: false,
      drawLongTick: false,
    },
    {
      value: 0.21428571428571427,
      label: '21',
      isWeekend: true,
      drawLongTick: false,
    },
    {
      value: 0.2857142857142857,
      label: '22',
      isWeekend: true,
      drawLongTick: false,
    },
    {
      value: 0.35714285714285715,
      label: '23',
      isWeekend: false,
      drawLongTick: false,
    },
    {
      value: 0.42857142857142855,
      label: '24',
      isWeekend: false,
      drawLongTick: false,
    },
    { value: 0.5, label: '25', isWeekend: false, drawLongTick: false },
    {
      value: 0.5714285714285714,
      label: '26',
      isWeekend: false,
      drawLongTick: false,
    },
    {
      value: 0.6428571428571429,
      label: '27',
      isWeekend: false,
      drawLongTick: false,
    },
    {
      value: 0.7142857142857143,
      label: '28',
      isWeekend: true,
      drawLongTick: false,
    },
    {
      value: 0.7857142857142857,
      label: '29',
      isWeekend: true,
      drawLongTick: false,
    },
    {
      value: 0.8571428571428571,
      label: '30',
      isWeekend: false,
      drawLongTick: false,
    },
    {
      value: 0.9285714285714286,
      label: '31',
      isWeekend: false,
      drawLongTick: false,
    },
  ],
  horizontalMonthTicks: [
    { value: 0, label: 'OCT / 2017', daysLeftInMonth: 14 },
  ],
  width: 1100,
  height: 350,
};
