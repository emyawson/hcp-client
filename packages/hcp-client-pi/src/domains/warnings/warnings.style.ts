import { IThemeInterface } from '@roche/patterns-indicators/theme';
import styled, { StyledComponentClass } from 'styled-components';

export const WarningsWrapper: StyledComponentClass<
  {},
  IThemeInterface
> = styled<{}, 'div'>('div')`
  width: 100%;
`;

WarningsWrapper.displayName = 'WarningsWrapper';

export const WarningsHeader: StyledComponentClass<{}, IThemeInterface> = styled<
  {},
  'div'
>('div')`
  width: 100%;
  display: flex;
  margin-bottom: ${props => props.theme.spacing.two};
`;

WarningsHeader.displayName = 'WarningsHeader';
