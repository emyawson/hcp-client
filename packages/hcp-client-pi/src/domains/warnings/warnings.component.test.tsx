import { shallow } from 'enzyme';
import * as React from 'react';

import { Warnings } from './warnings.component';

const warningProps = {
  selectedSegment: 'Trend Graph',
  selectSegment: () => null,
};

describe('Warnings component', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(<Warnings {...warningProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
