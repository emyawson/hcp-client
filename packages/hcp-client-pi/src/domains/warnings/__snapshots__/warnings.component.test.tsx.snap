// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warnings component Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    selectSegment={[Function]}
    selectedSegment="Trend Graph"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WarningsHeader>
          <Unknown
            labels={
              Array [
                "Trend Graph",
                "Log Book",
              ]
            }
            onSegmentClick={[Function]}
            selectedSegment="Trend Graph"
          />
        </WarningsHeader>,
        <Unknown
          height={350}
          horizontalDayTicks={
            Array [
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "18",
                "value": 0,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "19",
                "value": 0.07142857142857142,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "20",
                "value": 0.14285714285714285,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": true,
                "label": "21",
                "value": 0.21428571428571427,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": true,
                "label": "22",
                "value": 0.2857142857142857,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "23",
                "value": 0.35714285714285715,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "24",
                "value": 0.42857142857142855,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "25",
                "value": 0.5,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "26",
                "value": 0.5714285714285714,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "27",
                "value": 0.6428571428571429,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": true,
                "label": "28",
                "value": 0.7142857142857143,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": true,
                "label": "29",
                "value": 0.7857142857142857,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "30",
                "value": 0.8571428571428571,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "31",
                "value": 0.9285714285714286,
              },
            ]
          }
          horizontalMonthTicks={
            Array [
              Object {
                "daysLeftInMonth": 14,
                "label": "OCT / 2017",
                "value": 0,
              },
            ]
          }
          verticalLabel={
            Array [
              Object {
                "label": "Blood Glucose (mg/dL)",
                "value": 0.5,
              },
            ]
          }
          verticalTicks={
            Array [
              Object {
                "label": "0",
                "value": 0,
              },
              Object {
                "label": "100",
                "value": 0.25,
              },
              Object {
                "label": "200",
                "value": 0.5,
              },
              Object {
                "label": "300",
                "value": 0.75,
              },
              Object {
                "label": "400",
                "value": 1,
              },
            ]
          }
          width={1100}
        />,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown
            labels={
              Array [
                "Trend Graph",
                "Log Book",
              ]
            }
            onSegmentClick={[Function]}
            selectedSegment="Trend Graph"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "labels": Array [
              "Trend Graph",
              "Log Book",
            ],
            "onSegmentClick": [Function],
            "selectedSegment": "Trend Graph",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "height": 350,
          "horizontalDayTicks": Array [
            Object {
              "drawLongTick": false,
              "isWeekend": false,
              "label": "18",
              "value": 0,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": false,
              "label": "19",
              "value": 0.07142857142857142,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": false,
              "label": "20",
              "value": 0.14285714285714285,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": true,
              "label": "21",
              "value": 0.21428571428571427,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": true,
              "label": "22",
              "value": 0.2857142857142857,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": false,
              "label": "23",
              "value": 0.35714285714285715,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": false,
              "label": "24",
              "value": 0.42857142857142855,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": false,
              "label": "25",
              "value": 0.5,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": false,
              "label": "26",
              "value": 0.5714285714285714,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": false,
              "label": "27",
              "value": 0.6428571428571429,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": true,
              "label": "28",
              "value": 0.7142857142857143,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": true,
              "label": "29",
              "value": 0.7857142857142857,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": false,
              "label": "30",
              "value": 0.8571428571428571,
            },
            Object {
              "drawLongTick": false,
              "isWeekend": false,
              "label": "31",
              "value": 0.9285714285714286,
            },
          ],
          "horizontalMonthTicks": Array [
            Object {
              "daysLeftInMonth": 14,
              "label": "OCT / 2017",
              "value": 0,
            },
          ],
          "verticalLabel": Array [
            Object {
              "label": "Blood Glucose (mg/dL)",
              "value": 0.5,
            },
          ],
          "verticalTicks": Array [
            Object {
              "label": "0",
              "value": 0,
            },
            Object {
              "label": "100",
              "value": 0.25,
            },
            Object {
              "label": "200",
              "value": 0.5,
            },
            Object {
              "label": "300",
              "value": 0.75,
            },
            Object {
              "label": "400",
              "value": 1,
            },
          ],
          "width": 1100,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WarningsHeader>
            <Unknown
              labels={
                Array [
                  "Trend Graph",
                  "Log Book",
                ]
              }
              onSegmentClick={[Function]}
              selectedSegment="Trend Graph"
            />
          </WarningsHeader>,
          <Unknown
            height={350}
            horizontalDayTicks={
              Array [
                Object {
                  "drawLongTick": false,
                  "isWeekend": false,
                  "label": "18",
                  "value": 0,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": false,
                  "label": "19",
                  "value": 0.07142857142857142,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": false,
                  "label": "20",
                  "value": 0.14285714285714285,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": true,
                  "label": "21",
                  "value": 0.21428571428571427,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": true,
                  "label": "22",
                  "value": 0.2857142857142857,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": false,
                  "label": "23",
                  "value": 0.35714285714285715,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": false,
                  "label": "24",
                  "value": 0.42857142857142855,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": false,
                  "label": "25",
                  "value": 0.5,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": false,
                  "label": "26",
                  "value": 0.5714285714285714,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": false,
                  "label": "27",
                  "value": 0.6428571428571429,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": true,
                  "label": "28",
                  "value": 0.7142857142857143,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": true,
                  "label": "29",
                  "value": 0.7857142857142857,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": false,
                  "label": "30",
                  "value": 0.8571428571428571,
                },
                Object {
                  "drawLongTick": false,
                  "isWeekend": false,
                  "label": "31",
                  "value": 0.9285714285714286,
                },
              ]
            }
            horizontalMonthTicks={
              Array [
                Object {
                  "daysLeftInMonth": 14,
                  "label": "OCT / 2017",
                  "value": 0,
                },
              ]
            }
            verticalLabel={
              Array [
                Object {
                  "label": "Blood Glucose (mg/dL)",
                  "value": 0.5,
                },
              ]
            }
            verticalTicks={
              Array [
                Object {
                  "label": "0",
                  "value": 0,
                },
                Object {
                  "label": "100",
                  "value": 0.25,
                },
                Object {
                  "label": "200",
                  "value": 0.5,
                },
                Object {
                  "label": "300",
                  "value": 0.75,
                },
                Object {
                  "label": "400",
                  "value": 1,
                },
              ]
            }
            width={1100}
          />,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown
              labels={
                Array [
                  "Trend Graph",
                  "Log Book",
                ]
              }
              onSegmentClick={[Function]}
              selectedSegment="Trend Graph"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "labels": Array [
                "Trend Graph",
                "Log Book",
              ],
              "onSegmentClick": [Function],
              "selectedSegment": "Trend Graph",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 350,
            "horizontalDayTicks": Array [
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "18",
                "value": 0,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "19",
                "value": 0.07142857142857142,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "20",
                "value": 0.14285714285714285,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": true,
                "label": "21",
                "value": 0.21428571428571427,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": true,
                "label": "22",
                "value": 0.2857142857142857,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "23",
                "value": 0.35714285714285715,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "24",
                "value": 0.42857142857142855,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "25",
                "value": 0.5,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "26",
                "value": 0.5714285714285714,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "27",
                "value": 0.6428571428571429,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": true,
                "label": "28",
                "value": 0.7142857142857143,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": true,
                "label": "29",
                "value": 0.7857142857142857,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "30",
                "value": 0.8571428571428571,
              },
              Object {
                "drawLongTick": false,
                "isWeekend": false,
                "label": "31",
                "value": 0.9285714285714286,
              },
            ],
            "horizontalMonthTicks": Array [
              Object {
                "daysLeftInMonth": 14,
                "label": "OCT / 2017",
                "value": 0,
              },
            ],
            "verticalLabel": Array [
              Object {
                "label": "Blood Glucose (mg/dL)",
                "value": 0.5,
              },
            ],
            "verticalTicks": Array [
              Object {
                "label": "0",
                "value": 0,
              },
              Object {
                "label": "100",
                "value": 0.25,
              },
              Object {
                "label": "200",
                "value": 0.5,
              },
              Object {
                "label": "300",
                "value": 0.75,
              },
              Object {
                "label": "400",
                "value": 1,
              },
            ],
            "width": 1100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
